src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -346,8 +346,10 @@ class %s extends {$options['class']}
 
 EOF;
 
+                asort($this->preload);
+
                 foreach ($this->preload as $class) {
-                    if (!$class || false !== strpos($class, '$')) {
+                    if (!$class || false !== strpos($class, '$') || false === strpos($class, '\\')) {
```

Comment:
This will prevent preloading classes in the root namespace.
We should be able to use the ReflectionType::isBuiltin() method somewhere instead I suppose.

Reply:
Yes, I saw it in failed tests.

Reply:
I have no idea how to do it. First thought: exclusion list.

Reply:
The exclude list then.
Or keep as is since this has not side effect.